mino/minows/address.go

```diff
@@ -0,0 +1,156 @@
+package minows
+
+import (
+	"go.dedis.ch/dela"
+	"strings"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+	"go.dedis.ch/dela/mino"
+	"go.dedis.ch/dela/serde"
+	"golang.org/x/xerrors"
+)
+
+const protocolP2P = "/p2p/"
+
+// address represents a publicly reachable network address that can be used
+// to establish communication with a remote player through libp2p and,
+// therefore, must have both `location` and `identity` components.
+// - implements mino.Address
+type address struct {
+	location ma.Multiaddr
+	identity peer.ID
+}
+
+// newAddress creates a new address from a publicly reachable location with a
+// peer identity.
+func newAddress(location ma.Multiaddr, identity peer.ID) (address, error) {
+	if location == nil || identity.String() == "" {
+		return address{}, xerrors.New("address must have location and identity")
+	}
+	return address{
+		location: location,
+		identity: identity,
+	}, nil
+}
+
+// Equal implements mino.Address.
+func (a address) Equal(other mino.Address) bool {
+	switch o := other.(type) {
+	case address:
+		return equalOrBothNil(a.location, o.location) && a.identity == o.identity
+	case orchestratorAddr:
+		return o.Equal(a)
+	}
+	return false
+}
+
+// String implements fmt.Stringer.
+func (a address) String() string {
+	var sb strings.Builder
+
+	if a.location != nil {
+		sb.WriteString(a.location.String())
+	}
+
+	if a.identity.String() != "" {
+		sb.WriteString(protocolP2P)
+		sb.WriteString(a.identity.String())
```

Comment:
I can't remember `dela`'s convention right now, but shouldn't there be a separator of some type between these two/three values ?

Reply:
`minogrpc` uses a URL format in its [Address.String()](https://github.com/c4dt/dela/blob/main/mino/minogrpc/session/addr.go#L143) .

Here the `protocolP2P` is equal to `/p2p/`, so it's separated by a slash.

But I remember also one of the representations where it's just `/127.0.0.1/`.

So I'd say we're good here.